src/hooks/usePromiseDialog/PromiseDialogProvider.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
+import omit from 'lodash/omit';
+
+import {PromiseDialogContext} from './PromiseDialogContext';
+import type {DialogRendererProps, PromiseDialogResult} from './types';
+
+type PromiseDialogProviderProps = {
+    children: React.ReactNode | React.ReactNode[];
+};
+
+export const PromiseDialogProvider = ({children}: PromiseDialogProviderProps) => {
+    const [dialogs, setDialogs] = React.useState<Record<number, React.ReactNode>>([]);
+    const dialogsRef: React.MutableRefObject<Record<number, React.ReactNode>> =
+        React.useRef(dialogs);
+
+    React.useEffect(() => {
+        dialogsRef.current = dialogs;
+    }, [dialogs]);
```

Comment:
according to react useEffect usage recommendations better to use like this:
```tsx
if(dialogsRef.current !== dialogs){
  dialogsRef.current = dialogs;
};
```

Reply:
Thanks, I don't remember why it was done using an effect, since it's very old code)